### ISSUE TITLE
Feature request: A system to store players selected playertask

### ISSUE THREAD

**Issue by BillTheBeast** [Alt-Org/Altzone-Server#346]
## What do you want to see in the API?

Player should contain a playertask object that represents the selected task.
* This object just contains the id of the task and the progress of the task.
* Rest of the information can be fetched from the list received from the playertask endpoint.

Add POST playertasks/{_id} endpoint to receive information that a specific task has been chosen by the player
* Authentication token tells which player sent this message.
* Add this info to all the relevant locations. (ie. Player-data)
* If player already has a task selected return and error.

Add DELETE playertasks/{_id} endpoint to remove a selected task
* Mostly same as above

Add POST playerTask/progress that increments the progress of the task.
* Gets an integer that tell by how much should the task be incremented.
* If no value is provided we can maybe assume a value of one.
* Could also receive the id of the task to quarantee that the message is sent to the correct place.

If the task gets completed the rewards are distributed and the task is removed from the relevant locations.

**Comment by MikhailDeriabin** [comment]
@BillTheBeast 

The logic of player tasks is already handled by API. I do not understand why do you want to control the progress from client?

What is the chosen task of the player? The player has a specified set of tasks that need to be done, at least it was planned to work like that

**Comment by BillTheBeast** [comment]
> The logic of player tasks is already handled by API. I do not understand why do you want to control the progress from client?

Right, brainfart on my part. PlayerTasks can grab the progress from other server calls. 

> What is the chosen task of the player? The player has a specified set of tasks that need to be done, at least it was planned to work like that

Player can only have one task selected and they can only work on the selected task.

This idea has actually never changed and has been the plan for months now.
I also had to double check this from Helena.
But yes player can have many task to choose from, but only one can be selected at the time.

**Comment by MikhailDeriabin** [comment]
Ok, we will check it in the next sprint, but you can use the current system for now, its interface will not change

**Comment by MikhailDeriabin** [comment]
The task is split into smaller tasks, based on the meeting held 24.01.2025
